xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/compatibility.js

```diff
@@ -358,7 +358,101 @@ Object.extend(XWiki.constants, {
    * @deprecated since 4.2M1, your code shouldn't be aware of this separator, use {@code XWiki.Model}
    *             to resolve/serialize entity references
    */
-  pageAttachmentSeparator: "@"
+  pageAttachmentSeparator: "@",
+
+  /**
+   * Watchlist methods.
+   *
+   * @deprecated Since XWiki 7.4, the watchlist UI is implemented in a UI extension. This code is still there to not
+   * break the retro-compatibility but we can consider removing it.
+   */
+  watchlist : {
```

Comment:
Wondering how much that stuff shouldn't be move to the watch list extension code, instead of being put there.

Reply:
Moving it to the watch list extension is not very far from deleting it for good, as the code which stills uses this is not going to load it.